src/Utilities/SingleStackUtils/SingleStackUtils.jl

```diff
@@ -60,20 +62,31 @@ function get_vars_from_nodal_stack(
     stack_vals = OrderedDict()
     num_vars = varsize(vars)
     vars_wanted = Int[]
-    for vi in 1:num_vars
+    @inbounds for vi in 1:num_vars
         if !(var_names[vi] in exclude)
             stack_vals[var_names[vi]] = FT[]
             push!(vars_wanted, vi)
         end
     end
 
     # extract values from `state_data`
-    for ev in vrange
-        for k in 1:Nqk
+    @inbounds for ev in vrange, k in 1:Nqk, v in vars_wanted
+        if k == 1 && ev ≠ 1 && interp
+            ev⁻ = ev
+            ijk⁻ = i + Nq * ((j - 1) + Nq * (k - 1))
+
+            ev⁺ = ev - 1
+            ijk⁺ = i + Nq * ((j - 1) + Nq * (Nqk - 1))
+            state_local = state_data[ijk⁻, v, ev⁻]
+            state_local += state_data[ijk⁺, v, ev⁺]
+            state_local /= 2
+            push!(stack_vals[var_names[v]], state_local)
+        elseif k == Nqk && ev ≠ last(vrange) && interp
+            # Duplicate, already caught in above branch
+        else
```

Comment:
Something like this would work for non-periodic as well (if that's needed)

```suggestion
        if interp && k == 1
            # Get face degree of freedom number
            n = i + Nq * ((j - 1))
            # Get neighboring id data
            id⁻, id⁺ = vmap⁻[n, 5, ev⁻], vmap⁺[n, 5, ev⁻]
            # get the element numbers
            ev⁻ = ev
            ev⁺ = ((id⁺ - 1) ÷ Np) + 1
            # get the volume degree of freedom numbers
            vid⁻, vid⁺ = ((id⁻ - 1) % Np) + 1, ((id⁺ - 1) % Np) + 1

            J⁺, J⁻ = vgeo[vid⁻, _J, ev⁻], vgeo[vid⁺, _J, ev⁺]
            state_local = J⁻ * state_data[vid⁻, v, ev⁻]
            state_local += J⁺ * state_data[vid⁺, v, ev⁺]
            state_local /= (J⁻ + J⁺)
            push!(stack_vals[var_names[v]], state_local)
        elseif interp && k == Nqk
            # Get face degree of freedom number
            n = i + Nq * ((j - 1))
            # Get neighboring id data
            id⁻, id⁺ = vmap⁻[n, 6, ev⁻], vmap⁺[n, 6, ev⁻]
            # periodic and need to handle this point (otherwise handled above)
            if id⁺ == id⁻
                # get the element numbers
                ev⁻ = ev
                vid⁻ = ((id⁻ - 1) % Np) + 1

                state_local = state_data[vid⁻, v, ev⁻]
                push!(stack_vals[var_names[v]], state_local)
            end
        else
```

Reply:
Thanks! I'll try this out